src/NServiceBus.Core/Forwarding/ForwardingToDispatchConnector.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus
+{
+    using System;
+    using NServiceBus.Forwarding;
+    using NServiceBus.Pipeline;
+    using NServiceBus.TransportDispatch;
+    using NServiceBus.Transports;
+
+    class ForwardingToDispatchConnector:StageConnector<ForwardingContext,DispatchContext>
+    {
+        public override void Invoke(ForwardingContext context, Action<DispatchContext> next)
+        {
+            next(new DispatchContext(context.Get<OutgoingMessage>(),context));
```

Comment:
I like the idea but not sure I agree with the implementation. It seems a bit asymmetric since when forking we create a separate pipeline but when joining we add a connector. It would be symmetric if we either
 * have single outgoing `PipelineBase` instance but were able to route messages to multiple stages depending on context passed into next
 * have a separate `PipelineBase` instance for the dispatch pipeline (i.e. have proper dispatch pipeline instead of dispatch stage)